frontend/pages/donations/make.tsx

```diff
@@ -0,0 +1,118 @@
+import Container from "@material-ui/core/Container";
+import DropIn from "braintree-web-drop-in-react";
+import { GetServerSideProps } from "next";
+import { FC, useEffect, useState } from "react";
+import axiosWithoutAuth from "../../axiosWithoutAuth";
+import Layout from "../../components/Layout";
+
+interface DonateProps {
+  clientToken: string;
+}
+
+const Donate: FC<DonateProps> = (props: DonateProps) => {
+  const [clientToken, setClientToken] = useState(props.clientToken);
+  const [instance, setInstance] = useState(null);
+  useEffect(() => {
+    async function getClientToken() {
+      await axiosWithoutAuth
+        .get("/api/donations/token/")
+        .then((response) => {
+          setClientToken(response.data);
+        })
+        .catch((error) => {
+          console.error(error);
+        });
+      return {
+        props: {
+          clientToken,
+        },
+      };
+    }
+    getClientToken();
+  });
+
+  const makeDonation = (e) => {
+    e.preventDefault();
+    console.log("Triggered.");
+    async function getNonce() {
+      const { nonce } = await instance.requestPaymentMethod();
+      try {
+        // Send the nonce to the server.
+        const response = await axiosWithoutAuth.post("/api/donations/process/", {
+          paymentMethodNonce: nonce,
+          amount: $("#amount").val(),
+          name: $("#name").val(),
+        });
+        if (response.data == "OK") {
+          location.href = "/donations/success";
+        } else {
+          location.href = "/donations/error";
+        }
+      } catch (err) {
+        console.error(err);
+      }
+    }
+    getNonce();
+  };
+
+  return (
+    <Layout title="Donate">
+      <Container>
+        {(!clientToken && <p className="lead">Loading...</p>) || (
+          <div>
+            <header>
+              <p className="h3">Help us keep ModularHistory running</p>
+              <p>
+                ModularHistory provide its content for free. If you will like to help keep
+                ModularHistory afloat, you can use this form to make a donation.
+              </p>
+            </header>
+            <div className="row col-12 p-0 m-0 mt-3">
+              <div className="col pl-0">
+                <input
+                  type="text"
+                  className="form-control"
+                  name="name"
+                  id="name"
+                  placeholder="Name"
+                  required
+                />
+              </div>
+              <div className="col pr-0">
+                <input
+                  type="number"
+                  className="form-control"
+                  name="amount"
+                  id="amount"
+                  placeholder="Amount"
+                  required
+                />
+              </div>
+            </div>
+            <DropIn options={{ authorization: clientToken }} onInstance={(_) => setInstance(_)} />
```

Comment:
@sunzae99, let's add more options to DropIn.

Documentation for the DropIn component says we can pass any braintree dropin options to the `options` param of this component (see https://github.com/Cretezy/braintree-web-drop-in-react#options-object-required).

Braintree's documentation for dropin options can be found here:
https://braintree.github.io/braintree-web-drop-in/docs/current/module-braintree-web-drop-in.html#.create

Based on https://developers.braintreepayments.com/guides/drop-in/setup-and-integration/javascript/v3#paypal, I think that in order to add PayPal (in addition to credit card) as a payment option, we could probably do this:
```
<DropIn options={{ authorization: clientToken, paypal: {flow: "vault"} }} onInstance={(_) => setInstance(_)} />
```

Adding Venmo should be even easier; see:
https://developers.braintreepayments.com/guides/drop-in/setup-and-integration/javascript/v3#venmo

Reply:
Thanks for the references, @iacobfred. I will look into some more payment options that can be integrated into our existing payment methods.